core/src/main/java/io/questdb/cairo/DatabaseSnapshotAgentImpl.java

```diff
@@ -390,7 +408,12 @@ void recoverSnapshot() {
             memFile.smallFile(ff, srcPath, MemoryTag.MMAP_DEFAULT);
 
             final CharSequence currentInstanceId = configuration.getSnapshotInstanceId();
-            final CharSequence snapshotInstanceId = memFile.getStr(0);
+            CharSequence snapshotInstanceId = memFile.getStr(0);
+            if (Chars.empty(snapshotInstanceId)) {
+                srcPath.trimTo(snapshotRootLen).concat(TableUtils.SNAPSHOT_META_FILE_NAME_TXT).$();
+                snapshotInstanceId = TableUtils.readText(ff, srcPath);
```

Comment:
Might be a good idea to trim trailing spaces and newlines from the read text.

Reply:
Added .trim()